display/display.go

```diff
@@ -352,6 +353,42 @@ func Init(version, commit, builtBy string) {
 			case config.CmdAddSub:
 				go addSubscription()
 				return nil
+			case config.CmdCopyPageURL:
+				currentURL := tabs[curTab].page.URL
+				err := clipboard.WriteAll(currentURL)
+				if err != nil {
+					Error("Copy Error", err.Error())
+					return nil
+				}
+				return nil
+			case config.CmdCopyTargetURL:
+				currentURL := tabs[curTab].page.URL
+				selectedURL := tabs[curTab].HighlightedURL()
+				// Handle absolute URLs
+				if strings.HasPrefix(selectedURL, "gemini") || strings.HasPrefix(selectedURL, "http") {
+					err := clipboard.WriteAll(selectedURL)
+					if err != nil {
+						Error("Copy Error", err.Error())
+						return nil
+					}
+					return nil
+				} else if strings.HasPrefix(selectedURL, "/") {
+					// Handle relative URLs
+					u, _ := url.Parse(currentURL)
+					err := clipboard.WriteAll(u.Scheme + "://" + u.Host + selectedURL)
```

Comment:
Good thinking handling absolute and relative URLs. However there are problems with this approach. For example, if the selected URL is `mailto:test@example.com`. Or if the selected URL is relative but doesn't start with a slash, like `files/test.png`.

Instead of this, you should parse the `currentURL`, and then use [`.Parse`](https://golang.org/pkg/net/url/#URL.Parse) to get the absolute URL. If there's an error, just copy the text of the `selectedURL`.

Reply:
Good call. Updated to make better use of the Go standard library.